controller/src/main/java/org/jboss/as/controller/ParallelBootOperationContext.java

```diff
@@ -434,4 +436,14 @@ Resource getModel() {
     void logAuditRecord() {
         // handled by the primary context
     }
+
+    @Override
+    boolean isSkipModelValidation() {
+        return primaryContext.isSkipModelValidation();
+    }
+
+    @Override
+    void addModelValidationSteps() {
+        primaryContext.addModelValidationSteps();
```

Comment:
This will result in the steps getting added multiple times, as each subsystem runs, as the primary context is maintaining a global modifiedResourcesForModelValidation set.

It probably isn't causing problems because it's not getting called during boot, since skipModelValidation is true. But it's fragile. If it shouldn't ever be called perhaps its better to have it throw UnsupportedOperationException.